### PR TITLE
implement full coadd

### PR DESCRIPTION
**Pull request by esheldon**
untested

### CODE REVIEW

File: descwl_coadd/coadd.py [hunk 1]
```diff
@@ -1,16 +1,196 @@
...
+            exp = afw_image.ExposureF(masked_image)
+
+            exp_psf = make_stack_psf(psf_image)
+
+            exp.setPsf(exp_psf)
+
+            # set single WCS
+            crval = geom.SpherePoint(
+                wcs.center.ra/coord.radians,
+                wcs.center.dec/coord.radians,
```
Comment (beckermr) [inline descwl_coadd/coadd.py]: you can use `wcs.center.ra.deg` here I think.

File: descwl_coadd/coadd.py [hunk 2]
```diff
@@ -1,16 +1,196 @@
...
+                geom.radians,
+            )
+            stack_wcs = makeSkyWcs(
+                crpix=crpix,
+                crval=crval,
```
Comment (beckermr) [inline descwl_coadd/coadd.py]: I think you you will need `crpix+1` here since the sims come zero indexed

File: descwl_coadd/coadd.py [hunk 3]
```diff
@@ -1,16 +1,196 @@
...
+        self._weights = weight_list
+        self._input_recorder = input_recorder
+        self._coadd_psf_config = coadd_psf_config
+
+    def _make_coadd(self):
+        input_recorder = self._input_recorder
+
+        # combine stack images using mean
+        stats_flags = afw_math.stringToStatisticsProperty("MEAN")
+        stats_ctrl = afw_math.StatisticsControl()
```
Comment (beckermr) [inline descwl_coadd/coadd.py]: Does the stack know to convert to surface brightness?

File: descwl_coadd/coadd.py [hunk 4]
```diff
@@ -1,16 +1,206 @@
...
+
+            exp_psf = make_stack_psf(psf_image)
+
+            exp.setPsf(exp_psf)
+
+            # set single WCS
+            crval = geom.SpherePoint(
+                wcs.center.ra/coord.radians,
+                wcs.center.dec/coord.radians,
+                geom.radians,
```
Comment (beckermr) [inline descwl_coadd/coadd.py]: or I guess geom.degrees?

File: descwl_coadd/coadd.py (hunk 1 above)
Comment (beckermr) [reply descwl_coadd/coadd.py]: ack I had it in my head backwards what you are doing. ignore this.

File: descwl_coadd/coadd.py (hunk 4 above)
Comment (beckermr) [reply descwl_coadd/coadd.py]: ignore this too.

File: descwl_coadd/coadd.py (hunk 3 above)
Comment (esheldon) [reply descwl_coadd/coadd.py]: I have no idea

**Comment by esheldon** [comment]
some epochs

image-i-07

image-r-06



final coadd from 30 such epochs (10 in 3 bands)



coadd

**Comment by esheldon** [comment]
I'm going to merge this one and start more focused PRs

**Pull request merged**